pkg/subctl/operator/brokercr/ensure.go

```diff
@@ -0,0 +1,77 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package brokercr
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/retry"
+
+	submarinerv1a1 "github.com/submariner-io/submariner-operator/apis/submariner/v1alpha1"
+	submarinerclientset "github.com/submariner-io/submariner-operator/pkg/client/clientset/versioned"
+)
+
+const (
+	BrokerName = "submariner-broker"
+)
+
+func Ensure(config *rest.Config, namespace string, brokerSpec submarinerv1a1.BrokerSpec) error {
+	brokerCR := &submarinerv1a1.Broker{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: BrokerName,
+		},
+		Spec: brokerSpec,
+	}
+
+	clientSet, err := submarinerclientset.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	_, err = createBroker(clientSet, namespace, brokerCR)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createBroker(clientSet submarinerclientset.Interface, namespace string, brokerCR *submarinerv1a1.Broker) (bool, error) {
```

Comment:
Per @tpantelis I recently used the admiral util CreateOrUpdate, which resulted in more compact code.

https://github.com/submariner-io/submariner-operator/pull/1087/files#diff-8d3de7853ec49b469f54faa23b887aaea257e63e6cf1d5bfb891de5c2813685cR65

It has the disadvantage of needing the dynamic client, but the advantage that we don't need a different createOrUpdate per api type :)

Reply:
In this case though I don’t want to create or update, I want to create or delete then create.

Reply:
Fair enough